storage/badger/computation_result.go

```diff
@@ -0,0 +1,59 @@
+package badger
+
+import (
+	"errors"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger/operation"
+)
+
+type ComputationResultUploadStatus struct {
+	db *badger.DB
+}
+
+func NewComputationResultUploadStatus(db *badger.DB) *ComputationResultUploadStatus {
+	return &ComputationResultUploadStatus{
+		db: db,
+	}
+}
+
+func (c *ComputationResultUploadStatus) Upsert(executionDataId flow.Identifier,
```

Comment:
I get that its easier to retry if we store executionDataID, but would be great to key it by blockID, so its more consistent with rest of a system.

Reply:
addressed. the data retrieval interface now parameterized with a target status to filter